src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/createroles/CreateRoleAssignmentServiceTest.java

```diff
@@ -390,6 +437,108 @@ void identifyRoleAssignments_FromExistingRequest() {
         assertEquals(existingRecords.values(), sut.needToDeleteRoleAssignments.values());
     }
 
+    @Test
+    void checkUpdateExistingAssignmentNoMatchingID() throws IOException {
+
+
+        incomingAssignmentRequest = TestDataBuilder.buildAssignmentRequest(CREATED, LIVE,
+                                                                           false
+        );
+        Set<HistoryEntity> historyEntities = new HashSet<>();
+
+        historyEntities.add(historyEntity);
+
+        Map<UUID, RoleAssignmentSubset> needToDeleteRoleAssignments = new HashMap<>();
+
+        RoleAssignmentSubset roleAssignmentSubset = RoleAssignmentSubset.builder().build();
+        needToDeleteRoleAssignments.put(UUID.randomUUID(), roleAssignmentSubset);
+
+        requestEntity.setHistoryEntities(historyEntities);
+        sut.setRequestEntity(requestEntity);
+        sut.setNeedToDeleteRoleAssignments(needToDeleteRoleAssignments);
+
+        // actual Method call
+        sut.updateExistingAssignments(incomingAssignmentRequest);
+
+        //assertion
+        assertEquals(0,incomingAssignmentRequest.getRequestedRoles().size());
+        assertEquals(2,sut.needToRetainRoleAssignments.size());
+
+        verify(persistenceService, times(1))
+            .updateRequest(any(RequestEntity.class));
+        verify(persistenceService, times(1))
+            .persistHistoryEntities(any());
+
+    }
+
+    @Test
+    void checkUpdateNewAssignments() throws IOException, InvocationTargetException, IllegalAccessException {
+
+        incomingAssignmentRequest = TestDataBuilder.buildAssignmentRequest(CREATED, REJECTED,
+                                                                           false
+        );
+        existingAssignmentRequest = TestDataBuilder.buildAssignmentRequest(CREATED, REJECTED,
+                                                                           false
+        );
+        //prepare request entity
+        requestEntity = TestDataBuilder.buildRequestEntity(existingAssignmentRequest.getRequest());
+
+        sut.setRequestEntity(requestEntity);
+
+        //build history entity
+        historyEntity = TestDataBuilder.buildHistoryIntoEntity(
+            existingAssignmentRequest.getRequestedRoles().iterator().next(), requestEntity);
+
+        //set history entity into request entity
+        Set<HistoryEntity> historyEntities = new HashSet<>();
+        historyEntities.add(historyEntity);
+        requestEntity.setHistoryEntities(historyEntities);
+
+        RoleAssignmentSubset roleAssignmentSubset = RoleAssignmentSubset.builder().build();
+        roleAssignmentSubset
```

Comment:
Create a separate method to build RoleAssignmentSubset Object by the builder pattern rather than using the roleAssignmentSubset variable every time to set its attributes.

Reply:
Implemented